### ISSUE TITLE
`runtime-cast-value (TC006)` shouldn't apply to stubs

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#15176]
Found this whilst applying various Ruff groups on typeshed.

According to the updated typing spec for typing enums (see [Mypy: Change to Enum Membership Semantics](https://mypy-lang.blogspot.com/2024/12/mypy-114-released.html) and https://typing.readthedocs.io/en/latest/spec/enums.html#defining-members, this is the recommended way to type an enum in stubs where the type is known but the value isn't:
```py
class Key(enum.Enum):
    alt = cast(KeyCode, ...)
    alt_l = cast(KeyCode, ...)
    alt_r = cast(KeyCode, ...)
    alt_gr = cast(KeyCode, ...)
    # ...
```
This will trigger `runtime-cast-value (TC006)`

Related: https://github.com/astral-sh/ruff/issues/15132

Ruff: `0.8.4`
Command: `ruff check --preview --select=TC006 --isolated`

**Comment by Daverball** [comment]
My bad, some of the other new `flake8-type-checking` rules I added might also currently active in stub files. None of them should really be active.
